src/templates/parseJavaTemplates.ts

```diff
@@ -0,0 +1,192 @@
+    id: string;
+    metadata: {
+        defaultFunctionName: string;
+        name: string;
+        language: ProjectLanguage;
+        userPrompt?: string[];
+        bindingType: string;
+    };
+    files: { [filename: string]: string };
+}
+
+/**
+ * Describes a script template setting before it has been parsed
+ */
+interface IRawSetting {
+    name: string;
+    value: ValueType;
+    label: string;
+    defaultValue?: string;
+    required: boolean;
+    resource?: ResourceType;
+    validators?: {
+        expression: string;
+        errorText: string;
+    }[];
+    // tslint:disable-next-line:no-reserved-keywords
+    enum?: {
+        value: string;
+        display: string;
+    }[];
+}
+
+/**
+ * Describes script template config to be used for parsing
+ */
+interface IConfig {
+    variables: IVariables;
+    bindings: {
+        // tslint:disable-next-line:no-reserved-keywords
+        type: string;
+        settings: object[];
+    }[];
+}
+
+/**
+ * Describes script template variables to be used for parsing
+ */
+interface IVariables { [name: string]: string; }
+
+/**
+ * Describes script template resources to be used for parsing
+ */
+interface IResources { en: { [key: string]: string }; }
+
+const basicJavaTemplateNames: string[] = [
+    'HttpTrigger',
+    'BlobTrigger',
+    'QueueTrigger',
+    'TimerTrigger'
+];
+
+function parseScriptTemplate(rawTemplate: IRawTemplate, resources: IResources, commonSettings: IConfig): IFunctionTemplate {
+    const commonSettingsMap: { [inBindingType: string]: IFunctionSetting[] | undefined } = {};
+    for (const binding of commonSettings.bindings) {
+        commonSettingsMap[binding.type.toLowerCase()] = binding.settings.map((setting: object) => parseSetting(setting, resources, commonSettings.variables));
+    }
+
+    const userPromptedSettings: IFunctionSetting[] = [];
+    if (rawTemplate.metadata.userPrompt) {
+        for (const settingName of rawTemplate.metadata.userPrompt) {
+            const settings: IFunctionSetting[] | undefined = commonSettingsMap[rawTemplate.metadata.bindingType.toLowerCase()];
+            if (settings) {
+                const setting: IFunctionSetting | undefined = settings.find((bs: IFunctionSetting) => bs.name === settingName);
+                if (setting) {
+                    userPromptedSettings.push(setting);
+                }
+            }
+        }
+    }
+
+    return {
+        isHttpTrigger: rawTemplate.id.toLowerCase().startsWith('httptrigger'),
+        id: rawTemplate.id,
+        name: getResourceValue(resources, rawTemplate.metadata.name),
+        defaultFunctionName: rawTemplate.metadata.defaultFunctionName,
+        language: ProjectLanguage.Java,
+        userPromptedSettings: userPromptedSettings,
+        categories: [TemplateCategory.Core]
+    };
+}
+
+function parseSetting(data: object, resources: IResources, variables: IVariables): IFunctionSetting {
+    const rawSetting: IRawSetting = <IRawSetting>data;
```

Comment:
This file has a lot of code copied from `parseScriptTemplates` which we should leverage directly. For now, I suggest you just put `parseJavaTemplates` inside the same file as `parseScriptTemplates`. We should refactor it to a different file eventually (perhaps a separate PR right after this), but that will make it easier to review